flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/utils/MySqlTypeUtils.java

```diff
@@ -0,0 +1,186 @@
+            "DOUBLE PRECISION UNSIGNED ZEROFILL";
+    private static final String NUMERIC = "NUMERIC";
+    private static final String NUMERIC_UNSIGNED = "NUMERIC UNSIGNED";
+    private static final String NUMERIC_UNSIGNED_ZEROFILL = "NUMERIC UNSIGNED ZEROFILL";
+    private static final String FIXED = "FIXED";
+    private static final String FIXED_UNSIGNED = "FIXED UNSIGNED";
+    private static final String FIXED_UNSIGNED_ZEROFILL = "FIXED UNSIGNED ZEROFILL";
+    private static final String DECIMAL = "DECIMAL";
+    private static final String DECIMAL_UNSIGNED = "DECIMAL UNSIGNED";
+    private static final String DECIMAL_UNSIGNED_ZEROFILL = "DECIMAL UNSIGNED ZEROFILL";
+    private static final String CHAR = "CHAR";
+    private static final String VARCHAR = "VARCHAR";
+    private static final String TINYTEXT = "TINYTEXT";
+    private static final String MEDIUMTEXT = "MEDIUMTEXT";
+    private static final String TEXT = "TEXT";
+    private static final String LONGTEXT = "LONGTEXT";
+    private static final String DATE = "DATE";
+    private static final String TIME = "TIME";
+    private static final String DATETIME = "DATETIME";
+    private static final String TIMESTAMP = "TIMESTAMP";
+    private static final String YEAR = "YEAR";
+    private static final String BINARY = "BINARY";
+    private static final String VARBINARY = "VARBINARY";
+    private static final String TINYBLOB = "TINYBLOB";
+    private static final String MEDIUMBLOB = "MEDIUMBLOB";
+    private static final String BLOB = "BLOB";
+    private static final String LONGBLOB = "LONGBLOB";
+    private static final String JSON = "JSON";
+    private static final String SET = "SET";
+    private static final String ENUM = "ENUM";
+    private static final String GEOMETRY = "GEOMETRY";
+    private static final String UNKNOWN = "UNKNOWN";
+
+    /** Returns a corresponding Flink data type from a debezium {@link Column}. */
+    public static DataType fromDbzColumn(Column column) {
+        DataType dataType = convertFromColumn(column);
+        if (column.isOptional()) {
+            return dataType;
+        } else {
+            return dataType.notNull();
+        }
+    }
+
+    /**
+     * Returns a corresponding Flink data type from a debezium {@link Column} with nullable always
+     * be true.
+     */
+    private static DataType convertFromColumn(Column column) {
+        String typeName = column.typeName();
+        switch (typeName) {
+            case TINYINT:
+                // MySQL haven't boolean type, it uses tinyint(1) to represents boolean type
+                // user should not use tinyint(1) to store number although jdbc url parameter
+                // tinyInt1isBit=false can help change the return value, it's not a general way
+                // btw: mybatis and mysql-connector-java map tinyint(1) to boolean by default
+                return column.length() == 1 ? DataTypes.BOOLEAN() : DataTypes.TINYINT();
```

Comment:
We can introduce a `treat-tinyint1-as-boolean` config for mysql data source, a `TINYINT (1)` type in mysql can store value `2`,`3` too.